### PR TITLE
JSX type definition in Preact 8 is moved into preact namespace

### PR DESCRIPTION
**Pull request by 38elements**
This is related to #1811.

### CODE REVIEW

File: src/preact.d.ts [hunk 1]
```diff
@@ -154,18 +154,7 @@ declare namespace preact {
...
-type Defaultize<Props, Defaults> =
-	// Distribute over unions
-	Props extends any
-		? 	// Make any properties included in Default optional
-			& Partial<Pick<Props, Extract<keyof Props, keyof Defaults>>>
-			// Include the remaining properties from Props
-			& Pick<Props, Exclude<keyof Props, keyof Defaults>>
-		: never;
 
-declare global {
```
Comment (ForsakenHarmony) [inline src/preact.d.ts]: removing it would probably be a breaking change, I guess we'll need both and the only way to do it might be to copy paste

**Comment by 38elements** [comment]
@ForsakenHarmony 
Thank you for reviewing.
I changed this.

**Comment by ForsakenHarmony** [comment]
cc @pmkroeker

**Comment by pmkroeker** [comment]
Looks good to me!

**Comment by ForsakenHarmony** [comment]
Released in v8.5.0

**Comment by 38elements** [comment]
@ForsakenHarmony @pmkroeker 
Thanks.

**Pull request merged**